.github/workflows/publish.yml

```
@@ -0,0 +1,34 @@
+name: Publish to PyPi
+
+on:
+  push:
+    tags:
+      # This pattern is not a typical regular expression, see:
+      # https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet
+      - "v*"
+
+permissions:
+  contents: read
+
+jobs:
+  deploy:
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Setup poetry
+        id: poetry_setup
+        uses: ItsDrike/setup-poetry@v1
+        with:
+          python-version: 3.11
```

We probably need `poetry-dynamic-versioning` installed before running publish, so that poetry actually knows which version to publish

```suggestion
      - name: Setup poetry
        id: poetry_setup
        uses: ItsDrike/setup-poetry@v1
        with:
          python-version: 3.11
          install-args: "--with release"
```